auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonAdapterFactoryProcessor.java

```diff
@@ -135,15 +137,24 @@ private TypeSpec createTypeAdapterFactory(
         .returns(result)
         .addStatement("Class<$T> rawType = (Class<$T>) $N.getRawType()", t, t, type);
 
-    for (int i = 0, elementsSize = elements.size(); i < elementsSize; i++) {
-      Element element = elements.get(i);
+    List<Map.Entry<Element, Optional<ExecutableElement>>> properties = elements.stream()
+        .collect(Collectors.toMap(e -> e, this::getTypeAdapterMethod))
+        .entrySet()
+        .stream()
```

Comment:
collect to map, entry set, and stream all can be replaced by mapping to a pair / entry set directly.

Reply:
as in `elements.stream.map(e -> Pair(e, getTypeAdapterMethod(e)).collect(toList())`?

Reply:
Yep!

Reply:
I'll need to find a Pair implementation I can use here, oddly not even guava has it in its kitchen sink :|. Might just handwrite in a little holder here

Reply:
e4b8b79

Reply:
`AbstractMap.SimpleEntry`

Reply:
Mind if I just keep the pair that I added here? Technically it doesn't really need to be a map anyway, I was just sheepishly using it to get an entry set 😁